api/src/main/java/org/ccci/idm/user/User.java

```diff
@@ -447,6 +457,53 @@ public void setTelephoneNumber(String telephoneNumber) {
         this.telephoneNumber = telephoneNumber;
     }
 
+    public String getSecurityQuestion() {
+        return securityQuestion;
+    }
+
+    public void setSecurityQuestion(final String securityQuestion) {
+        this.securityQuestion = securityQuestion;
+    }
+
+    /**
+     * Not meant for public use.
+     */
+    public String getSecurityAnswer() {
+        return securityAnswer;
+    }
+
+    /**
+     * Check the plain text security answer against this security answer, which is already hashed
+     *
+     * @param securityAnswer plain text security answer
+     *
+     * @return true if provided security answer matches this object's
+     */
+    public boolean checkSecurityAnswer(final String securityAnswer) {
+        return !Strings.isNullOrEmpty(this.securityAnswer) &&
+                HashUtility.checkHash(normalize(securityAnswer), this.securityAnswer);
+    }
+
+    /**
+     * Store the hashed value of the plain text security answer
+     *
+     * @param securityAnswer plain text security answer
+     */
+    public void setSecurityAnswer(final String securityAnswer) {
+        this.setSecurityAnswer(securityAnswer, true);
+    }
+
+    /**
+     * Not meant for public use.
+     */
+    public void setSecurityAnswer(final String securityAnswer, boolean hash) {
+        this.securityAnswer = hash ? HashUtility.getHash(normalize(securityAnswer)) : securityAnswer;
+    }
+
+    private String normalize(final String string) {
+        return CharMatcher.WHITESPACE.trimAndCollapseFrom(string, ' ').toLowerCase();
```

Comment:
good call on the collapse as well, I didn't realize that existed in `CharMatcher`